### ISSUE TITLE
List Formatting

### ISSUE THREAD

**Issue by chulbert** [ZhgChgLi/ZMarkupParser#52]
This is an excellent library that meets all our needs except one: list formatting.

For an unordered list like the following:

```
<ul>
  <li>ZMarkupParser is a pure-Swift library that helps you convert HTML strings into NSAttributedString with customized styles and tags.</li>
  <li>ZMarkupParser is a pure-Swift library that helps you convert HTML strings into NSAttributedString with customized styles and tags.</li>
  <li>ZMarkupParser is a pure-Swift library that helps you convert HTML strings into NSAttributedString with customized styles and tags.</li>
</ul>
```

The rendered output is:

![Screenshot 2024-03-15 at 11 26 14 AM](https://github.com/ZhgChgLi/ZMarkupParser/assets/9003435/3b893a27-0286-4512-b7ea-bf52b3c2211b)

This falls short of our need in a couple ways:

1. There is no spacing between the bullet and the text.  This makes them very hard to read.
2. The wrapped text should align with the start of the first line, similar to the following:

<img width="355" alt="Screenshot 2024-03-15 at 11 35 41 AM" src="https://github.com/ZhgChgLi/ZMarkupParser/assets/9003435/5165a69f-93a5-4fc2-9500-ae74cbce5ea2">

It would be great if this could be achieved.  Otherwise lists are fairly unusable.

**Comment by pykaso** [comment]
You can take a look at my [PR](https://github.com/ZhgChgLi/ZMarkupParser/pull/60), where I am trying to solve this problem.

**Comment by zhgchgli0718** [comment]
Thanks @chulbert for reporting this issue and @pykaso for providing the solution PR.
Based on the PR provided by @pykaso, I gained a lot of inspiration regarding HTML list formatting.
As a result, I studied the entire problem and proposed a more comprehensive fix and solution.
Thank you all for 

![Simulator Screenshot - iPhone 15 - 2024-05-28 at 17 45 56](https://github.com/ZhgChgLi/ZMarkupParser/assets/33706588/1f9ca0de-891c-464e-be5c-be5b7d6660a8)
your support and patience.
